google/cloud/internal/grpc_opentelemetry.cc

```diff
@@ -30,6 +32,41 @@ namespace internal {
 
 #ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
 
+namespace {
+
+/**
+ * A [carrier] for gRPC.
+ *
+ * This code originates from the following example:
+ *
+ * https://github.com/open-telemetry/opentelemetry-cpp/blob/a343da043e1351c3cc3003bfce94724345ee22f1/examples/grpc/tracer_common.h#L26-L45
+ *
+ * [carrier]:
+ * https://opentelemetry.io/docs/reference/specification/context/api-propagators/#carrier
+ */
+class GrpcClientCarrier
+    : public opentelemetry::context::propagation::TextMapCarrier {
+ public:
+  explicit GrpcClientCarrier(grpc::ClientContext& context)
+      : context_(context) {}
+
+  // Unneeded by clients.
+  opentelemetry::nostd::string_view Get(
+      opentelemetry::nostd::string_view) const noexcept override {
+    return "";
+  }
+
+  void Set(opentelemetry::nostd::string_view key,
+           opentelemetry::nostd::string_view value) noexcept override {
+    context_.AddMetadata(key.data(), value.data());
+  }
```

Comment:
Using `sv.data()` without `sv.size()` is a red flag.

Reply:
Ah, TIL. Thanks for checking and good catch.

I think that explains why I lost hours of time to this piece of code: https://github.com/dbolduc/google-cloud-cpp/blob/9e9691476a167626486dba50efab0a7eafef2334/google/cloud/trace/propagator/cloud_trace_context.cc#L62